src/main/java/pl/mantiscrab/budgetr/registration/InMemoryUserRepository.java

```diff
@@ -1,22 +1,21 @@
 package pl.mantiscrab.budgetr.registration;
 
+import org.springframework.dao.DataIntegrityViolationException;
+import pl.mantiscrab.budgetr.InMemoryDummyCrudRepository;
+
 import java.util.Objects;
 import java.util.Optional;
 
-class InMemoryUserRepository extends InMemoryDummyCrudRepository<User, Long> implements UserRepository{
+class InMemoryUserRepository extends InMemoryDummyCrudRepository<User, String> implements UserRepository{
```

Comment:
What is the purpose of having both h2 and this repositories?

Reply:
I don't understand the question. In memory repositories are for unit tests, and H2 for integration